### DISCUSSION TITLE
❌ Full Shell Access

### DISCUSSION THREAD

**Discussion by LostRhoades** [isontheline/pro.webssh.net#590] [Ideas]
WebSSH is great.  I'm so glad it's being actively supported.  What are the chances of getting full shell access (Mac OS)?  I believe there is a restriction in the Apple Store which doesn't allow this, but I've seen other Apps provide a separate download option from another site, which is a more full featured version of the App.  Just an idea.  Thanks!

**Comment by isontheline** [comment]
Hello @LostRhoades 👋

Thank you a lot for your support about WebSSH 🙏

You're right, the sandbox doesn't allow WebSSH to run external parts.
The only way as you said is to package WebSSH outside the App Store.

Which parts are you missing from mashREPL? 
Command line apps or the ability to access your full hard drive?

**Reply by LostRhoades** [reply]
I've seen what is available in mashREPL and there are some real helpful tools there.  It would be nice to have direct access to my hard drive.  What would be ideal would be 'one client to rule them all'.  Sort of a perfect blend of WebSSH and the native MacOS Terminal.

**Reply by isontheline** [reply]
Thank you a lot for your feedback 👍

Other app you are using is Termius?

**Reply by LostRhoades** [reply]
I've tried Terminus.  It's great and well built, but their subscription is ridiculous.  I could never justify that.  Before I ever found WebSSH I found PortX.  That's the app that I was referring to that has a 'download outside the appstore' option.  It's far from perfect and still a work in progress, but it's a decent app.  I like the tab system, the built in shell and the ability to split it into multiple windows within the same screen.  My biggest grip with it is the bookmarks/groups.  Things aren't alphabetically sorted, it's a static ordering which you can't change, so it doesn't scale/grow well.

**Reply by isontheline** [reply]
Thanks for sharing your though about PortX.
I think also that it will be a great app in a few weeks.
There are so many great SSH apps nowadays.

I will try to do my best effort in order to offer an "out of store" release, supporting native terminal apps.

**Reply by LostRhoades** [reply]
That would be fantastic.  I realize you have a lot of requests, so I appreciate you looking into it.

**Reply by isontheline** [reply]
All enhancements need to be looked at 😉
I love to learn about all use cases in order to smooth everyone workflows.

**Reply by isontheline** [reply]
@LostRhoades As a workaround you can add a SSH connection to your own localhost device. So you will have full access to your system and apps.

**Reply by LostRhoades** [reply]
That's thinking outside the box.  I'd never considered that.  Thanks for the suggestion!
